packages/nft-erc721-provider/lib/NftErc721Provider.ts

```diff
@@ -0,0 +1,64 @@
+import { addressToString } from '@liquality/utils'
+import { ensure0x, normalizeTransactionObject } from '@liquality/ethereum-utils'
+import { Address } from '@liquality/types'
+import { NftBaseProvider } from '@liquality/nft-base-provider'
+
+import { Signer } from '@ethersproject/abstract-signer'
+
+import NftErc721_ABI from './NftErc721_ABI.json'
+const erc721InterfaceID = '0x80ac58cd'
+
+export default class NftErc721Provider extends NftBaseProvider {
+  constructor(signer: Signer) {
+    super(signer, NftErc721_ABI as any, erc721InterfaceID)
+  }
+
+  async balance(contract: Address | string) {
+    await super.balance(contract)
+
+    const amount = await this._contract.functions.balanceOf(await this._signer.getAddress())
+    return amount[0].toNumber()
+  }
+
+  async transfer(contract: Address | string, receiver: Address | string, tokenID: number) {
```

Comment:
Since this nft standard does not let you transfer batch, I think its important to make `transfer` for single and anohter method `transferBatch` for multiple. 

A user may add this provider and then call `client.nft.transfer` with multiple tokenIDs and expect that it should work

Reply:
The idea here is that the end user/dev would only use the `NftProvider` without knowing about the different standards (721 & 1155).